cli/internal/git/pull.go

```diff
@@ -50,5 +64,31 @@ func pull(gitUrl string, targetFolder string) {
 		logContext.Fatal("Not a valid git repo or unable to clone")
 	}
 
+	if !fetchAllTags {
+		tag := matches[1]
+
+		// Identify the remote trunk branch name
+		trunkBranchName := plumbing.NewBranchReferenceName("master")
```

Comment:
I redid this a little, I've had issue in golang before doing `err == nil` and try to always use `err != nil` because of past issues, https://yourbasic.org/golang/gotcha-why-nil-error-not-equal-nil/.  The nested conditionals were a little hard to follow.  Most of this is subjective, but I always strive to reduce cognitive load when possible.  Feel free to change some more if I messed this up.

Reply:
That looks good to me. I wasn't a huge fan of the nested conditional either, was just trying to avoid this:
```go
if err != nil {
    logContext.Debug(err)
    logContext.Warn(message)
} else if !head.Name().IsBranch() {
    logContext.Warn(message)
} else {
    trunkBranchName = head.Name()
}
```

since the `if` and `else if` warning log are the same message. With how you have the messages setup I like this much better.

Reply:
That's very interesting with the whole `nil != nil` thing. Thanks for the share!

Reply:
Yeah I had to play with the logic a little to get it how I thought it might read better.  I also did different warns since technically they are different conditions occurring between the err and not being a branch.